front/src/applications/operationalStudies/components/MacroEditor/export.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  osrdEditoastApi,
+  type Distribution,
+  type TrainScheduleBase,
+} from 'common/api/osrdEditoastApi';
+import type { AppDispatch } from 'store';
+
+import type { NGEDeleteEvent, NGEEvent, NGEUpdateEvent, OperationTrainrunSection } from './types';
+
+// TODO: add a dynamic values when available
+const DEFAULT_PAYLOAD = {
+  constraint_distribution: 'STANDARD' as Distribution,
+  rolling_stock_name: '',
+  start_time: '2024-07-15T08:00:00+02:00',
+};
+
+const createPayloadOperation = (trainrunSections: OperationTrainrunSection[]) => {
+  // TODO: check that the trainrunSections format is still compatible
+  const path = trainrunSections.flatMap((section, index) => {
+    const [sourceTrigram, sourceSecondaryCode] = section.sourceNode.betriebspunktName.split('/');
+    const [targetTrigram, targetSecondaryCode] = section.targetNode.betriebspunktName.split('/');
+    const sourceNode = {
+      trigram: sourceTrigram,
+      secondary_code: sourceSecondaryCode,
+      id: `${section.sourceNode.id}-${index}`,
+    };
+    const targetNode = {
+      trigram: targetTrigram,
+      secondary_code: targetSecondaryCode,
+      id: `${section.targetNode.id}-${index}`,
+    };
+    if (index === trainrunSections.length - 1) {
+      return [sourceNode, targetNode];
+    }
+    return [sourceNode];
+  });
+  const [firstTrainrunSection] = trainrunSections;
+  const payloads: TrainScheduleBase = {
+    ...DEFAULT_PAYLOAD,
+    path,
+    train_name: firstTrainrunSection.trainrun.name,
+  };
+  return payloads;
+};
+
+const handleCreateOperation = (trainrunSection: OperationTrainrunSection) => [
+  createPayloadOperation([trainrunSection]),
+];
+
+const handleUpdateOperation = (trainrunSections: OperationTrainrunSection[]) =>
+  createPayloadOperation(trainrunSections);
+
+const ngeOperations = async (event: NGEEvent, dispatch: AppDispatch, timeTableId: number) => {
+  switch (event.type) {
+    case 'create': {
+      await dispatch(
+        osrdEditoastApi.endpoints.postV2TimetableByIdTrainSchedule.initiate({
+          id: timeTableId,
+          body: handleCreateOperation(event.trainrunSection),
+        })
+      ).unwrap();
+      break;
+    }
+    case 'delete': {
+      const { trainrun } = event as NGEDeleteEvent;
+      await dispatch(
+        osrdEditoastApi.endpoints.deleteV2TrainSchedule.initiate({ body: { ids: [trainrun.id] } })
+      ).unwrap();
+      break;
+    }
+    case 'update': {
+      const { trainrunSections } = event as NGEUpdateEvent;
+      const [firstTrainrunSection] = trainrunSections;
```

Comment:
Maybe we should add a test here to check that `trainrunSections.length > 0`, otherwise the app will panick l76
```suggestion
      const [firstTrainrunSection] = trainrunSections;
```

Reply:
I believe this is guaranteed by NGE. Wouldn't hurt to add a `console.assert()` or throw an error out of caution though.